### ISSUE TITLE
Improve Accessibility of Inputs (Date/Category/Location)

### ISSUE THREAD

**Issue by djamell** [DNNCommunity/DNN.Events#215]
## Is your feature request related to a problem?
No, just an enhancement.

The Events Module remarkably passes almost every Accessibility test except the Input controls at the top of the module.  When I test using the  WebAIM Chrome Extension (WAVE) I get errors on all three  (Date/Category/Location) controls.

The errors relate to missing Form Labels, Empty Buttons, and Empty Table Headers.

## Describe the solution you'd like

I would like for the controls to be tweaked so they pass WAVE.

**Comment by EPTamminga** [comment]
I will give this some attention for the next release.
BTW a screen shot with some marking would be helpful.

**Comment by djamell** [comment]
![WAVE](https://user-images.githubusercontent.com/312711/74878213-7d0ea680-532c-11ea-8435-60ed16b70f34.png)

**Comment by djamell** [comment]
Just a gentle nudge to keep this Issue alive.

**Comment by djamell** [comment]
Another bump to keep this Issue alive.

**Comment by CubeDweller** [comment]
Another bump.
Did this even get resolved?

**Comment by valadas** [comment]
Unfortunately no contribution came in for this issue. If anyone intends to submit a code contribution or sponsor some development for it, feel free to comment and I will be glad to reopen. Stalebot closed it due to inactivity

**Comment by djamell** [comment]
I'm the original poster and I'd be happy to sponsor the change if it's not too expensive.

**Comment by valadas** [comment]
Ok, let's start with pinging @EPTamminga as I believe he is the one that knows the most about this module codebase...

@djamell  If you don't hear back in a few days, please comment again and we can see who else can jump in (maybe myself if it's not too rush)...

**Comment by djamell** [comment]
OK, will do.  No real rush.  Thank you Daniel!

**Comment by EPTamminga** [comment]
I have been maintaining this module since 2007, with various people contributing code. 
It started when needed this module back in those days and there were some errors I fixed.  
I do not use this module with any projects we have done in the last 5 yrs or so, As a result, the urgency to enhance or fix things have gone down.. Sorry.

I am not looking for money, If you want to donate anything, please consider becoming a member of the DNNCommunity team on KIVA, https://www.kiva.org/team/dnn_community (if you are not yet) and start with micro financing to people who really need it.

I can do some work, but I can use some detailed instructions on what has to be done.

**Comment by djamell** [comment]
HI EP.  Thanks for your willingness to do this.  In part, you can refer to the early posts in made in this thread.

Specifically, the three input controls (Date/Category/Location) do not have labels so screen readers can't tell what the comtrol does.  The error is described by the WAVE as the following:

**What It Means**

A form control does not have a corresponding label.

**Why It Matters**

If a form control does not have a properly associated text label, the function or purpose of that form control may not be presented to screen reader users. Form labels also provide visible descriptions and larger clickable targets for form controls.

**How to Fix It**

If a text label for a form control is visible, use the <label> element to associate it with its respective form control. If there is no visible label, either provide an associated label, add a descriptive title attribute to the form control, or reference the label(s) using aria-labelledby. Labels are not required for image, submit, reset, button, or hidden form controls.

**The Algorithm... in English**

An <input> (except types of image, submit, reset, button, or hidden), <select>, or <textarea> does not have a properly associated label. A properly associated label is:
a <label> element with a for attribute value that is equal to the id of a unique form control
a <label> element that surrounds the form control, does not surround any other form controls, and does not reference another element with its for attribute
a non-empty title attribute, or
a non-empty aria-labelledby attribute.

**Comment by EPTamminga** [comment]
AFAIK, there are (localizable) labels connected to the controls in the standard DNN way.
I will try to have a look next week.

**Comment by EPTamminga** [comment]
@djamell I have been analyzing it, I thought I found the solution, by specifying the ID of the in[ut control in the label control. That does did not seem to work correctly.
* The date input is a Telerik RAD editor and the input textbox gets a generated name
* The combo boxes for location and category are DNNCombobox controls. I linked the control ID to the label (this ID link was missing in the label control), but that did not seem to work correctly.

I had a look at other parts of DNN, eg. the standard settings pages (tabbed page) and also there, some of the controls get a marker from WAVE, others do not. It remains unclear to me why one is OK and the other not.

![image](https://user-images.githubusercontent.com/5859855/109838136-2d083680-7c46-11eb-8bf6-c198159ecc75.png)

![image](https://user-images.githubusercontent.com/5859855/109838172-342f4480-7c46-11eb-9f06-3500dfb5c641.png)

The "good news" is that the Telerik RAD date control has to be replaced by a none Telerik control I assume that that will solve the case for the date field.

For the DNNCombo controls: I am not sure about how to solve that yet.
